### PR TITLE
Installer error handling improvements and auto gen random conf

### PR DESCRIPTION
**Pull request by hack-beard**
Added more verbose error handling to installer script and added in a function to populate the api_key and crypt_salt fields in default.conf with random uuids. Need to do additional testing on centos and Darwin to verify, but no issues expected

### CODE REVIEW

**Comment by privateducky** [comment]
this looks good. let me know when you've confirmed against non-Windows machines

File: install.sh [hunk 1]
```diff
@@ -3,103 +3,162 @@
...
+function initialize_log() {
+    echo "CALDERA install log">install_log.txt
+}
+
+function extra_error() {
+    if [[ $? == 0 ]]; then
+        echo "[+] $2 installed"
+        echo "[+] $2 installed">>install_log.txt
+    else
+        echo "[x] $2 FAILED to install"
```
Comment (wbooth) [inline install.sh]: use variable to eliminate duplicate strings

**Comment by hack-beard** [comment]
@wbooth should be resolved

**Comment by hack-beard** [comment]
Tested on centos. lsb-release was not installed by default on my centos, i can add in the function to install redhat-lsb-core on centos, but the way the script is structured at the moment, it will not run until after the script decides which OS it is on, which requires the lsb-release functionality. Recommend scoping out this fault for the time being and revisiting in the future if script is re-structured. May just be an artifact of my install

**Comment by hack-beard** [comment]
@wbooth resolved "sed" and sudo check on Darwin. Still issued with "readlink" on darwin, which prevents script from pulling the caldera directory. Current thought is to replace this with a simple pwd command and assume user is running script from same directory, not sure if this is acceptable change

**Pull request merged**